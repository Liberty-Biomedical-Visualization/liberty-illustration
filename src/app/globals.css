@tailwind base;
@tailwind components;
@tailwind utilities;

:root {
  --accent-50: 224 247 252;
  --accent-100: 178 224 240;
  --accent-200: 128 200 232;
  --accent-300: 77 176 215;
  --accent-400: 26 153 206;
  --accent-500: 3 152 206;
  --accent-600: 2 128 165;
  --accent-700: 2 109 136;
  --accent-800: 2 90 107;
  --accent-900: 2 61 77;
  --accent-950: 1 44 58;
  --background: 255 255 255;
  --foreground: 23 23 23;
}

body {
  color: var(--foreground);
  background: var(--background);
  font-family: Arial, Helvetica, sans-serif;
}

@layer utilities {
  .text-balance {
    text-wrap: balance;
  }
}
