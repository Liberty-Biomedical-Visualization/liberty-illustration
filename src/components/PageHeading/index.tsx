import resolveClassNames from "@/lib/resolveClassNames";

export default function PageHeading(props: Readonly<PageHeadingProps>) {
  const { children } = props;
  const className = resolveClassNames(baseClassName, props.className);
  return <h2 className={className}>{children}</h2>;
}

const baseClassName = "sm:text-3xl text-2xl text-center";

export interface PageHeadingProps {
  children: string;
  className?: string;
}
