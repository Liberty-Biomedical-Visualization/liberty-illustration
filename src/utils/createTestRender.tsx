import { render } from "@testing-library/react";
import type { JSX, JSXElementConstructor } from "react";

/**
 * Returns a function that allows rendering Component with partial props while
 * falling back to defaultProps for testing purposes.
 */
export default function createTestRender<C extends JSXElementConstructor<P>, P>(
  Component: JSXElementConstructor<P> & C,
  defaultProps: JSX.LibraryManagedAttributes<C, P>,
) {
  return function (props: Partial<P> = {}) {
    const resolvedProps: JSX.IntrinsicAttributes &
      JSX.LibraryManagedAttributes<C, P> = { ...defaultProps, ...props };
    return render(<Component {...resolvedProps} />);
  };
}
