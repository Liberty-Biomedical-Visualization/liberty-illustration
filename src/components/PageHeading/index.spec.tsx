import { describe, expect, it } from "@jest/globals";
import { screen } from "@testing-library/react";

import PageHeading from ".";
import createTestRender from "@/lib/createTestRender";

describe("PageHeading", () => {
  it("should display an level two heading", () => {
    const children = "Medical Illustration";
    renderPageHeading({ children });
    const result = screen.getByRole("heading", { level: 2, name: children });
    expect(result).toBeInTheDocument();
  });

  it("should pass className to the outermost element", () => {
    const className = "foo";
    const result = renderPageHeading({ className });
    expect(result.container.firstChild).toHaveClass(className);
  });

  it("should match the snapshot", () => {
    const result = renderPageHeading();
    expect(result.container).toMatchSnapshot();
  });
});

const renderPageHeading = createTestRender(PageHeading, {
  children: "Medical Illustration",
});
