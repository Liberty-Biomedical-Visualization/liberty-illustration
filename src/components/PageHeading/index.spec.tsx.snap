// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeading should match the snapshot 1`] = `
<div>
  <h2
    class="sm:text-3xl text-2xl text-center"
  >
    Medical Illustration
  </h2>
</div>
`;
